src/query/functions/utils/heap.go

```diff
@@ -137,14 +149,27 @@ func (fh *FloatHeap) Reset() {
 }
 
 // Flush flushes the float heap and resets it. Does not guarantee order.
-func (fh FloatHeap) Flush() []ValueIndexPair {
+func (fh *FloatHeap) Flush() []ValueIndexPair {
 	values := fh.floatHeap.heap
 	fh.Reset()
 	return values
 }
 
+// FlushInOrder flushes the float heap and returns values in order.
+func (fh *FloatHeap) FlushInOrder() []ValueIndexPair {
+	result := make([]ValueIndexPair, len(fh.floatHeap.heap))
+
+	for i := len(result) - 1; i >= 0; i-- {
+		if e, ok := fh.Pop(); ok {
+			result[i] = e
+		}
+	}
```

Comment:
```suggestion
	result := make([]ValueIndexPair, 0, fh.Len())

	for e, ok := fh.Pop(); ok {
	  result = append(result, e)
	}
```

Reply:
Your suggestion doesn't compile and semantics are different here. 
There is a reason why we need to iterate backwards, it's because `Pop()` will return items in the opposite order (e.g. for max heap it will retain the largest values in the heap, but Pop() will return smallest value).